source/common/ssl/context_config_impl.cc

```diff
@@ -39,26 +39,33 @@ ContextConfigImpl::ContextConfigImpl(const envoy::api::v2::CommonTlsContext& con
       ecdh_curves_(StringUtil::nonEmptyStringOrDefault(
           RepeatedPtrUtil::join(config.tls_params().ecdh_curves(), ":"), DEFAULT_ECDH_CURVES)),
       ca_cert_file_(config.validation_context().trusted_ca().filename()),
-      cert_chain_file_(config.tls_certificates()[0].certificate_chain().filename()),
-      private_key_file_(config.tls_certificates()[0].private_key().filename()),
+      cert_chain_file_(config.tls_certificates().size()
```

Comment:
nit: Please use `.empty()` in the various places you are basing logic off of size() being 0 or not zero.

Reply:
Done.